docs/src/dev-docs/new-architecture.rst

```diff
@@ -0,0 +1,146 @@
+        checkpoints_dir="path",
+    )
+
+    model.save_checkpoint("final.ckpt")
+
+    mts_atomistic_model = model.export()
+    mts_atomistic_model.export("path", collect_extensions="extensions-dir/")
+
+
+In order to follow this, a new architectures has two define two classes
+
+- a ``Model`` class, defining the core of the architecture. This class must implement
+  the interface documented below in :py:class:`ModelInterface`
+- a ``Trainer`` class, used to train an architecture and produce a model that can be
+  evaluated and exported. This class must implement the interface documented below in
+  :py:class:`TrainerInterface`.
+
+The ``ModelInterface`` is the main model class and must implement a
+``save_checkpoint()``, ``load_checkpoint()``  as well as a ``restart()`` and
+``export()`` method.
+
+.. code-block:: python
+
+    class ModelInterface:
+        def __init__(self, architecture_hypers, dataset_info: DatasetInfo):
+            pass
+
+        def save_checkpoint(self, path):
+            pass
+
+        @classmethod
+        def load_checkpoint(cls, path) -> "ModelInterface":
+            pass
+
+        def restart(self, dataset_info: DatasetInfo) -> "ModelInterface":
+            """
+            restart training
+
+            This function is called whenever training restarts, with the same or a
+            different dataset.
+
+            It enables transfer learning (changing the targets), and fine tuning (same
+            targets, different dataset)
+            """
+            pass
+
+        def export(self) -> MetatensorAtomisticModel:
+            pass
+
+Note that
+the ``ModelInterface`` does not necessary inherit from :py:class:`torch.nn.Module` since
+training can be performed in any way.
+
+The ``export()`` method is required to transform a trained model into a standalone file
+to be used in combination with molecular dynamic engines to run simulations. We provide
+a helper function :py:func:`metatensor.models.utils.export.export` to export a torch
+model to an :py:class:`MetatensorAtomisticModel
+<metatensor.torch.atomistic.MetatensorAtomisticModel>`.
+
+The ``TrainerInterface`` class should have the following signature with a required
+methods for ``train()``.
+
+.. code-block:: python
+
+    class TrainerInterface:
+        def __init__(self, train_hypers):
+            pass
+
+        def train(
+            self,
+            model: ModelInterface,
+            devices: List[torch.device],
+            train_datasets: List[Union[Dataset, torch.utils.data.Subset]],
+            validation_datasets: List[Union[Dataset, torch.utils.data.Subset]],
+            checkpoints_dir: str,
+        ): ...
+
+The names of the ``ModelInterface`` and the ``TrainerInterface`` are free to choose but
+should be linked to constants in the ``__init__.py`` of each architecture. On top of
+these two constants the ``__init__.py`` must contain constants for the
+`__capabilities__`, the `__authors__` and
+`__maintainers__`.
```

Comment:
Missing `__name__` or `__architecture_name__`

Reply:
I though we decided this was not required, since it can be deduced from the path of the file.